entity-framework/core/what-is-new/ef-core-8.0/breaking-changes.md

```diff
@@ -437,3 +440,96 @@ This was done to further improve startup time.
 #### Mitigations
 
 Edit the generated `*ModelBuilder.cs` file and remove the line `AddRuntimeAnnotation("Relational:RelationalModel", CreateRelationalModel());` as well as the method `CreateRelationalModel()`.
+
+<a name="navigation-names"></a>
+
+### Scaffolding may generate different navigation names
+
+[Tracking Issue #27832](https://github.com/dotnet/efcore/issues/27832)
+
+#### Old behavior
+
+Previously when scaffolding a `DbContext` and entity types from an existing database, the navigation names for relationships were sometimes derived from a common prefix of multiple foreign key column names.
+
+#### New behavior
+
+Starting with EF Core 8.0, common prefixes of column names from a composite foreign key are no longer used to generate navigation names.
+
+#### Why
+
+This is an obscure naming rule which sometimes generates very poor names like, `S`, `Student_`, or even just `_`. Without this rule, strange names are no longer generated, and the naming conventions for navigations are also made simpler, thereby making it easier to understand and predict which names will be generated.
+
+#### Mitigations
+
+The [EF Core Power Tools](https://github.com/ErikEJ/EFCorePowerTools/issues/2143) have an option to keep generating navigations in the old way. Alternatively, the code generated can be fully customized using [T4 templates](xref:core/managing-schemas/scaffolding/templates). This can be used to example the foreign key properties of scaffolding relationships and use whatever rule is appropriate for your code to generate the navigation names you need.
+
+<a name="discriminators"></a>
+
+### Discriminators now have a max length
```

Comment:
That's not really the break we're trying to document. The migration is an appropriate way to manage the change. However, this is not the case when Migrations is not able to automatically make the change, such as when the column is part of an index, since the index would need to dropped and re-created, and EF doesn't do that, so it fails.

Reply:
I see, ok. It still may be worth explaining that an unexpected migration is created (and why) as background etc. But no big deal either way.

Reply:
Yeah, I added another sentence to that effect.